codegen/lib/templates/coins.cpp.erb

```diff
@@ -83,6 +83,60 @@ enum TWPublicKeyType TW::publicKeyType(TWCoinType coin) {
     }
 }
 
+byte TW::p2pkhPrefix(TWCoinType coin) {
+    switch (coin) {
+<%  coins.filter{ |coin| coin.has_key?('p2pkhPrefix') }.each do |coin| -%>
+    case TWCoinType<%= format_name(coin['name']) %>:
+        return <%= coin['p2pkhPrefix'] %>;
+<% end -%>
+    default: return 0;
+    }
+}
+
+std::set<byte> TW::p2pkhPrefixes() {
+    return std::set<byte>({
+<%  coins.filter{ |coin| coin.has_key?('p2pkhPrefix') && coin['id'] != 'decred' }
+        .map{ |coin| coin['p2pkhPrefix'] }
+        .uniq
+        .each do |prefix|
+-%>
+        <%= prefix %>,
+<% end -%>
+    });
+}
+
+byte TW::p2shPrefix(TWCoinType coin) {
+    switch (coin) {
+<%  coins.filter{ |coin| coin.has_key?('p2shPrefix') }.each do |coin| -%>
+    case TWCoinType<%= format_name(coin['name']) %>:
+        return <%= coin['p2shPrefix'] %>;
+<% end -%>
+    default: return 0;
+    }
+}
+
+std::set<byte> TW::p2shPrefixes() {
+    return std::set<byte>({
+<%  coins.filter{ |coin| coin.has_key?('p2shPrefix') && coin['id'] != 'decred' }
+        .map{ |coin| coin['p2shPrefix'] }
+        .uniq
+        .each do |prefix|
+-%>
+        <%= prefix %>,
+<% end -%>
+    });
+}
+
+enum TWHRP TW::hrp(TWCoinType coin) {
+    switch (coin) {
+<%  coins.filter{ |coin| coin.has_key?('hrp') }.each do |coin| -%>
```

Comment:
Any chance we can add test for it? in the case somehow break it

Reply:
to each function

Reply:
Yes, we have `TWHRPTests`, just make sure test new added hrp